sklearn/inspection/_plot/partial_dependence.py

```diff
@@ -0,0 +1,243 @@
+
+    axes_ : dict of matplotlib Axes
+        If `ax` is an axes or None, `axes_[i, j]` is the axes on the ith row
+        and jth column. If `ax` is a list of axes, `axes_[i]` is the ith item
+        in `ax`.
+
+    lines_ : matplotlib Artists
+        If `ax` is an axes or None, `line_[i, j]` is the partial dependence
+        curve on the ith row and jth column. If `ax` is a list of axes,
+        `lines_[i]` is the partial dependence curve corresponding to the ith
+        item in `ax`. Only plots with a single feature will be populated in
+        `lines_`. Contour plots are stored in `contours_`.
+
+    contours_ : matplotlib Artists
+        If `ax` is an axes or None, `contours_[i, j]` is the partial dependence
+        plot on the ith row and jth column. If `ax` is a list of axes,
+        `contours_[i]` is the partial dependence plot corresponding to the ith
+        item in `ax`. Only plots with a two feature will be populated in
+        `contours_`. Line plots are stored in `lines_`.
+
+    figure_ : matplotlib Figure
+        Figure containing partial dependence plots.
+    """
+    def __init__(self, pd_results, features, feature_names, target_idx,
+                 pdp_lim, deciles):
+        self.pd_results = pd_results
+        self.features = features
+        self.feature_names = feature_names
+        self.target_idx = target_idx
+        self.pdp_lim = pdp_lim
+        self.deciles = deciles
+
+    def plot(self, ax=None, n_cols=3, line_kw=None, contour_kw=None, fig=None):
+        """Plot partial dependence plots.
+
+        Parameters
+        ----------
+        ax : Matplotlib axes, list of Matplotlib axes or None, default=None
+            Axes to plot the partial dependence curves. If a single axes is
+            given, it is treated as a bounding axes and a grid of partial
+            depdendence plots will be drawn on that top of it. If a list of
+            axes are passed, the partial dependence plots will be drawn on
+            those axes. By default, a single bounding axes is created and
+            treated as the single axes case.
+
+        n_cols : int, default=3
+            The maximum number of columns in the grid plot.
+
+        line_kw : dict or None, default=None
+            Dict with keywords passed to the ``matplotlib.pyplot.plot`` call.
+            For one-way partial dependence plots.
+
+        contour_kw : dict or None, default=None
+            Dict with keywords passed to the ``matplotlib.pyplot.contourf``
+            call for two-way partial dependence plots.
+
+        fig : Matplotlib figure object or None, default=None
+            A figure object onto which the plots will be drawn, after the
+            figure has been cleared. By default, a new one is created.
+
+            .. deprecated:: 0.22
+
+        Returns
+        -------
+        display: :class:`~sklearn.inspection.PartialDependenceDisplay`
+        """
+
+        check_matplotlib_support("plot_partial_dependence")
```

Comment:
I would tend toward having a `sklearn.utils._plot._validate_mpl_import` and call that. (Trying to avoid inheritance for now)

What do you see `_validate_mpl_import` returning?